aep/general/0135/aep.md.j2

```diff
@@ -0,0 +1,314 @@
+  // (Otherwise, the request will only work if the publisher has no books.)
+  bool force = 2;
+}
+```
+
+The API **must** fail with a `FAILED_PRECONDITION` error if the `force` field
+is `false` (or unset) and child resources are present.
+
+{% endtabs %}
+
+### Protected delete
+
+Sometimes, it may be necessary for users to ensure that no changes have been
+made to a resource that is being deleted. If a resource provides an [etag][],
+the delete request **may** accept the etag (as either required or optional):
+
+{% tab oas -%}
+
+{% sample 'protected_delete.oas.yaml', 'paths' %}
+
+If the etag is provided and does not match the server-computed etag, the
+request **must** fail with `412 Precondition Failed`.
+
+{% tab proto -%}
+
+```proto
+message DeleteBookRequest {
+  // The name of the book to delete.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/Book"
+    }];
+
+  // Optional. The etag of the book.
+  // If this is provided, it must match the server's etag.
+  string etag = 2;
+}
+```
+
+If the etag is provided and does not match the server-computed etag, the
+request **must** fail with a `ABORTED` error code.
+
+{% endtabs %}
+
+**Note:** Declarative-friendly resources ([AEP-128][aep-128]) **must** provide the `etag`
+field for Delete requests.
+
+### Delete if existing
+
+If the service uses client-assigned resource names, `Delete` methods **may**
+expose a `bool allow_missing` field, which will cause the method to succeed in
+the event that the user attempts to delete a resource that is not present (in
+which case the request is a no-op):
+
+{% tab oas -%}
+
+{% sample 'delete_if_existing.oas.yaml', 'paths' %}
+
+More specifically, the `allow_missing` flag triggers the following behavior:
+
+- If the method call is on a resource that does not exist, the request is a
+  no-op.
+  - Conditional headers (e.g. `If-Match`) are ignored.
+- If the method call is on a resource that already exists, the resource is
+  deleted (subject to other checks).
+
+{% tab proto -%}
+
+```proto
+message DeleteBookRequest {
+  // The book to delete.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference).type = "library.googleapis.com/Book"
+  ];
+
+  // If set to true, and the book is not found, the request will succeed
+  // but no action will be taken on the server
+  bool allow_missing = 2;
+}
+```
+
+More specifically, the `allow_missing` flag triggers the following behavior:
+
+- If the method call is on a resource that does not exist, the request is a
+  no-op.
+  - The `etag` field is ignored.
+- If the method call is on a resource that already exists, the resource is
+  deleted (subject to other checks).
+
+{% endtabs %}
+
+**Note:** Declarative-friendly resources ([AEP-128][aep-128]) **should** expose the
```

Comment:
we can remove this guidance. We were working on removing AIP-128 (consolidating that guidance in the appropriate sections) at Google. I don't think there's a need to bring this over.